pkg/kubectl/cmd/util/factory.go

```diff
@@ -275,6 +275,49 @@ func makeInterfacesFor(versionList []unversioned.GroupVersion) func(version unve
 	}
 }
 
+type DiscoveryExpander struct {
+	ShortcutExpander
+	discoveryClient *discovery.DiscoveryClient
+}
+
+func (e DiscoveryExpander) getAll() []unversioned.GroupResource {
```

Comment:
Can you make this the impl in `ShortcutExpander`?  You could make `discoveryClient` optional, so people can still get fixed mappings.